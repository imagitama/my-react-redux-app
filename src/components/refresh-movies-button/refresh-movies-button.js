import React from 'react'
import { connect } from 'react-redux'
import styled from 'react-emotion'
import { fetchMovies } from '../../actions/movies'

const StyledRefreshMoviesButton = styled('button')({
  padding: '0.5rem',
  margin: '1rem auto',
  display: 'block'
})

export const RefreshMoviesButton = ({ fetchMovies }) => 
<StyledRefreshMoviesButton onClick={fetchMovies}>Refresh!</StyledRefreshMoviesButton>

export default connect(null, { fetchMovies })(RefreshMoviesButton)