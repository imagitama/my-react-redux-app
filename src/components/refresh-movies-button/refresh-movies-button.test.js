import React from 'react'
import { mount } from 'enzyme'
import { RefreshMoviesButton } from './refresh-movies-button'

it('Matches the snapshot', () => {
  expect(mount(<RefreshMoviesButton fetchMovies={() => null} />)).toMatchSnapshot()
})