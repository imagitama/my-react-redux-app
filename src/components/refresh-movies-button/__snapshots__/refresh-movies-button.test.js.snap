// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<Component
  fetchMovies={[Function]}
>
  <Styled(button)
    onClick={[Function]}
  >
    <button
      className="css-12fl8ck"
      onClick={[Function]}
    >
      Refresh!
    </button>
  </Styled(button)>
</Component>
`;
